packages/@n8n/nodes-langchain/nodes/mcp/McpClientTool/McpClientTool.node.ts

```diff
@@ -239,7 +231,22 @@ export class McpClientTool implements INodeType {
 			endpointUrl = this.getNodeParameter('endpointUrl', itemIndex) as string;
 		}
 
-		const { headers } = await getAuthHeaders(this, authentication);
+		if (authentication !== 'none') {
```

Comment:
stolen from the http request tool node, not sure if this check is also needed here